### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by naumenkogs**

### CODE REVIEW

File: contracts/ExtendedSimpleAuthorization.sol [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import "./SimpleAuthorization.sol";
+
+// Truffle overloading workaround
+contract ExtendedSimpleAuthorization is SimpleAuthorization {
+
+    function check2(address _token, address _address) public /* view */ returns (uint8 resultCode) {
+        return check(_token,_address);
+    }
+
+    function check4(address _token, address _from, address _to, uint256 _amount) public /* view */ returns (uint8 resultCode) {
```
Comment (expede) [inline contracts/ExtendedSimpleAuthorization.sol]: Why `check4` and `check2` with the explicit numbers? Is it idiomatic in Solidity?

File: test/spec/reference-token-spec.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+async function failTransaction(func, args, errorMessage) {
+  try {
+    await func.apply(this, args);
+    throw "Should have failed";
+  } catch (e) {
+    assert.equal(e.message, errorMessage);
+  }
+}
+
+revertMessage = "VM Exception while processing transaction: revert";
```
Comment (expede) [inline test/spec/reference-token-spec.js]: Does this need a `const`?

File: test/spec/reference-token-spec.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      balance = await referenceToken.balanceOf(targetAccount);
+      assert.equal(balance, amountToMint * granularity);
+  });
+
+  it('reference token receiver (mint) should be authorized', async () => {
+      targetAccount = accounts[1];
+      const amountToMint = 100;
+      await failTransaction(referenceToken.mint, [targetAccount, amountToMint * granularity], revertMessage);
+      await simpleAuthorization.setAuthorized(targetAccount, true);
+      await referenceToken.mint(targetAccount, amountToMint * granularity);
```
Comment (expede) [inline test/spec/reference-token-spec.js]: No `assert`? Or is the test that it blows up on failure?

File: contracts/ExtendedSimpleAuthorization.sol (hunk 1 above)
Comment (naumenkogs) [reply contracts/ExtendedSimpleAuthorization.sol]: This "Extended" contract is a workaround for truffle test framework not able to parse overloaded functions.

Comment (expede) [reply contracts/ExtendedSimpleAuthorization.sol]: Ooooh, right. I remember this now 😭

File: test/spec/reference-token-spec.js (hunk 3 above)
Comment (naumenkogs) [reply test/spec/reference-token-spec.js]: I'm not sure how can you do it with assert.

failTransaction is a wrapper for cases when EVM throws an error (revertMessage in this case helps to check that it is expected error)

Comment (expede) [reply test/spec/reference-token-spec.js]: @naumenkogs apologies, this has an `assert` in it. I was looking all over in the docs for `failTransaction`, but it's defined at the top of the file 😅 It has an assert in it:

`assert.equal(e.message, errorMessage)` 👍🏻

Comment (expede) [reply test/spec/reference-token-spec.js]: Would this be a helpful function to port to `web3-server-tools`? Think that we'll be using it elsewhere?

**Pull request merged**